### PR TITLE
Site Editor: Remove useEditedEntityRecord hook

### PR DESCRIPTION
**Pull request by youknowriad**
Related #66921 

## What?

As the site editor is growing to become a multi page app, it doesn't really make sense to have an edited post type and context state in it.
That state is forcing us today to have a two synchronization mechanism between the url and the state causing some issues related to performance and some hidden bugs at times.

Ideally the routes would just pass the post type, post id to the editor component. The current PR helps to go in that direction by removing the useEditedEntityRecord hook that used to rely on these selectors.

## Testing Instructions

- No functional change you can smoke test loading the site editor.

### CODE REVIEW

File: packages/edit-site/src/components/editor/index.js [hunk 1]
```diff
@@ -126,7 +126,10 @@ export default function EditSiteEditor( { isPostsList = false } ) {
 			hasSiteIcon: !! siteData?.site_icon_url,
 		};
 	}, [] );
-	useEditorTitle();
+	useEditorTitle(
+		postWithTemplate ? contextPostType : editedPostType,
+		postWithTemplate ? contextPostId : editedPostId
```
Comment (mcsf) [inline packages/edit-site/src/components/editor/index.js:132]: Is this actually working? `postWithTemplate` is only assigned after this call.

**Comment by mcsf** [comment]
I know that we've generally neglected edit-site's changelog, but maybe add these changes to it?

File: packages/edit-site/src/components/editor/index.js (hunk 1 above)
Comment (youknowriad) [reply packages/edit-site/src/components/editor/index.js:132]: good call.

**Comment by youknowriad** [comment]
> I know that we've generally neglected edit-site's changelog, but maybe add these changes to it?

I considered it but this is really an internal hook, so it felt unnecessary

**Review by Mamaduka — APPROVED** [review]
Nothing came up when smoke testing the changes 🚢

**Pull request merged**